src/main/java/org/shanerx/tradeshop/listeners/ShopProtectionListener.java

```diff
@@ -144,79 +142,53 @@ public void onEntityExplodeItem(EntityExplodeEvent event) {
+                if (destroyEvent.isCancelled()) {
+                    event.setCancelled(true);
+                    return;
                 }
 
+                shop.getChestAsSC().resetName();
                 shop.remove();
                 return;
             }
+
             event.setCancelled(true);
             player.sendMessage(Message.NO_TS_DESTROY.getPrefixed());
 
-        } else if (plugin.getListManager().isInventory(block)) {
-            BlockState bs = block.getState();
-            if (!ShopChest.isShopChest(bs.getBlock()))
-                return;
-
-            if (player.hasPermission(Permissions.ADMIN.getPerm())) {
-                new ShopChest(block.getLocation()).resetName();
-                return;
-            }
-
-            Sign s = findShopSign(block);
-            if (s == null) {
-                new ShopChest(block.getLocation()).resetName();
-                return;
-            }
-
-            if (!ShopType.isShop(s)) {
-                new ShopChest(block.getLocation()).resetName();
+        } else if (ShopChest.isShopChest(block)) {
+            shop = new ShopChest(block.getLocation()).getShop();
+            if (shop == null)
                 return;
-            }
+            if (player.hasPermission(Permissions.ADMIN.getPerm()) || player.getUniqueId().equals(shop.getOwner().getUUID())) {
+                PlayerShopDestroyEvent destroyEvent = new PlayerShopDestroyEvent(player, shop);
+                Bukkit.getPluginManager().callEvent(destroyEvent);
+                if (destroyEvent.isCancelled()) {
+                    event.setCancelled(true);
+                    return;
+                }
 
-            Shop shop = Shop.loadShop(s);
+                shop.getChestAsSC().resetName();
+                shop.removeStorage();
 
-            if (event.getPlayer().getUniqueId().equals(shop.getOwner().getUUID())) {
-	            PlayerShopDestroyEvent destroyEvent = new PlayerShopDestroyEvent(player, shop);
-	            Bukkit.getPluginManager().callEvent(destroyEvent);
-	            if (destroyEvent.isCancelled()) {
-		            event.setCancelled(true);
-		            return;
-	            }
-	            
-                if (!ShopChest.isDoubleChest(block)) {
-                    new ShopChest(shop.getInventoryLocation()).resetName();
-                    shop.removeStorage();
-                    shop.updateStatus();
-                    shop.saveShop();
+                if (shop.getShopSign() == null) {
+                    shop.remove();
                 } else {
-                    if (bs instanceof Nameable && ((Nameable) bs).getCustomName() != null
-                            && ((Nameable) bs).getCustomName().contains("$ ^Sign:l_")) {
-                        ((Nameable) bs).setCustomName(((Nameable) bs).getCustomName().split("\\$ \\^")[0]);
-
-                        bs.update();
-                    }
+                    shop.updateSign();
```

Comment:
what exactly is this for?

Reply:
updates the text on the sign when the chest is broken, including status and everything

Reply:
ty for clarifying